packages/cadl-ranch-specs/http/authentication/apiKey/mockapi.ts

```diff
@@ -0,0 +1,24 @@
+import { passOnSuccess, mockapi, json } from "@azure-tools/cadl-ranch-api";
+import { ScenarioMockApi } from "@azure-tools/cadl-ranch-api";
+
+export const Scenarios: Record<string, ScenarioMockApi> = {};
+
+Scenarios.ApiKeyAuthentication_ApiKey_authenticated = passOnSuccess(
+  mockapi.get("/authentication/api-key/authenticated", (req) => {
+    req.expect.containsHeader("x-ms-api-key", "valid-key");
+    return { status: 204 };
+  }),
+);
+
+Scenarios.ApiKeyAuthentication_ApiKey_invalidAuthentication = passOnSuccess(
+  mockapi.get("/authentication/api-key/invalid-authentication", (req) => {
+    return { status: 403,
```

Comment:
Since api-key is usually not via "Authentication" header, return 403 (instead of 401 which would require an 'WWW-Authenticate` header for challenge.

Checked storage and search, both uses 403 (though storage actually send the SharedKey via Authentication header).

---

Java azure-core actually forbid sending key on HTTP.
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/http/policy/AzureKeyCredentialPolicy.java#L33

Hence testing is done via Postman.

Reply:
We could make the mock api support https for completeness, probably would have to just do one of the following:
- Install the mock api certificate before calling. A bit anoying but would be the most accurate testing.
- Disable cert validation on that test. Usually much easier but I guess that depends on the language.

Reply:
Maybe we can delay the decision of whether to have HTTPS, depends on whether other language get to the problem.

If the problem is Java only, we probably do a less accurate testing by somehow disable that check in Cadl Java tests.